be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -137,157 +180,269 @@ object DbActor extends AskPatternConstants {
    */
   def apply(mediatorRef: ActorRef, DATABASE_FOLDER: String = "database"): DbActor = {
 
-    val CHANNELS_FOLDER: String = s"$DATABASE_FOLDER/channels"
-
     /* Create a logger for this DB actor"*/
     val logger = LoggerFactory.getLogger("DBLogger")
     
     val options: Options = new Options()
     options.createIfMissing(true)
-    val channelNamesDb: DB =
-      try { factory.open(new File(CHANNELS_FOLDER), options) }
+    
+    val db: DB =
+      try { factory.open(new File(DATABASE_FOLDER), options) }
       catch {
         case e: IOException => {
-          logger.error("Could not open channels folder {}", CHANNELS_FOLDER)
+          logger.error("Could not open database folder {}", DATABASE_FOLDER)
           throw e
         }
 
       }
-    val iterator: DBIterator = channelNamesDb.iterator
-    val initialChannelsMap: mutable.Map[Channel, DB] = mutable.Map.empty
-
-    iterator.seekToFirst()
-    options.createIfMissing(false)
-
-    while (iterator.hasNext) {
-      // open each db associated with each channel name
-      val channelName: String =
-        new String(iterator.next().getKey, StandardCharsets.UTF_8)
-      val channelDb: DB =
-        try { factory.open(new File(channelName), options) }
-        catch {
-          case e: IOException => {
-            logger.error("Could not open channel {}", channelName)
-            throw e
-          }
-
-        }
-
-      // store the channel name and its database in the map
-      initialChannelsMap += (Channel(
-        channelName.toString.stripPrefix(DATABASE_FOLDER)
-      ) -> channelDb)
-    }
-
-    iterator.close()
-    DbActor(mediatorRef, initialChannelsMap.toMap, channelNamesDb, DATABASE_FOLDER, CHANNELS_FOLDER)
+    DbActor(mediatorRef, db, DATABASE_FOLDER)
   }
 
-  sealed case class DbActor(mediatorRef: ActorRef, initialChannelsMap: Map[Channel, DB], channelNamesDb: DB, DATABASE_FOLDER: String, CHANNELS_FOLDER: String) extends Actor with ActorLogging {
-    private val channelsMap: mutable.Map[Channel, DB] = initialChannelsMap.to(collection.mutable.Map)
+  // In the arguments, we might remove the initialChannelsMap and replace the channelNamesDb by the general Db??
+  // FIXME: find way to keep count of channel nb. Idea: store key "channelCount" with value int.getBytes to db and increment it with each channel created
+  sealed case class DbActor(mediatorRef: ActorRef, db: DB, DATABASE_FOLDER: String) extends Actor with ActorLogging {
 
     override def preStart(): Unit = {
```

Comment:
I don't think this method should be kept at all. If of course there is no need to run some code before deploying the DBactor. More info here ` https://www.javatpoint.com/akka-actor-life-cycle`

Reply:
I'll remove this indeed